src/main/java/net/minecrell/serverlistplus/core/status/PersonalizedStatusPatch.java

```diff
@@ -70,63 +72,149 @@ public boolean hasChanges() {
     // Getters
     public Boolean hidePlayers(StatusResponse response) {
         Boolean result;
-        return response.getRequest().isIdentified() && (result = personalized.getHidePlayers()) != null ?
-                result : def.getHidePlayers();
+        if (response.getRequest().isIdentified()) {
+            if (response.getCore().getPlugin().isBanned(response.getRequest().getIdentity())
```

Comment:
You should add `isBanned(Core)` to `PlayerIdentity` and use that here, saves some code and looks nicer!